### PR TITLE
[fix][client] Fix for early hit `beforeConsume` for MultiTopicConsumer

### PR DESCRIPTION
**Pull request by coderzc**
### Motivation

When using MultiTopicConsumer with ConsumerInterceptor , the `beforeConsume` method was hit many times
before receiving the first message. The root cause of this is that before the MultiTopicsConsumer receives the message, MultiTopicsConsumer has already completed receiving the message from the internal consumer and executed the `beforeConsume` method, so we can let `beforeConsume` be called in MultiTopicsConsumer to avoid this issue.

https://github.com/apache/pulsar/blob/8707fbe8351fb6ac4337fbd88d86eb32aff55b04/pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java#L228-L254

### Modifications

* Call `beforeConsume` method in MultiTopicsConsumer.
* Wrap ConsumerInterceptors for internal consumer the  and remove `beforeConsume` logic

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

**Review by codelipenghui — COMMENTED** [review]
The fix for `beforeConsume` method looks good.
@coderzc Could you please help also check other methods in the ConsumerInterceptor to make sure they are working as expected without issues like `beforeConsume` method.

File: pulsar-broker/src/test/java/org/apache/pulsar/client/api/InterceptorsTest.java [hunk 1]
```diff
@@ -856,6 +855,91 @@ public void onPartitionsChange(String topicName, int partitions) {
         Assert.assertNull(reader.readNext(3, TimeUnit.SECONDS));
     }
 
+    @Test
+    public void testConsumerInterceptorWithMultiTopics() throws PulsarClientException {
```
Comment (codelipenghui) [inline pulsar-broker/src/test/java/org/apache/pulsar/client/api/InterceptorsTest.java:859]: We already have the test `testConsumerInterceptorWithMultiTopicSubscribe` but it looks like the test can't cover this case. It's better to update the existing one to cover this case without maintaining to similar tests.

Comment (coderzc) [reply pulsar-broker/src/test/java/org/apache/pulsar/client/api/InterceptorsTest.java:865]: Ok, improved test.

**Comment by coderzc** [comment]
> The fix for `beforeConsume` method looks good. @coderzc Could you please help also check other methods in the ConsumerInterceptor to make sure they are working as expected without issues like `beforeConsume` method.

I checked other methods in the ConsumerInterceptor, they work well.

**Comment by coderzc** [comment]
@codelipenghui PTAL~

**Review by shibd — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**